src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -352,47 +416,116 @@ public void ProcessTestRunAttachments(TestRunAttachmentsProcessingPayload attach
                     EqtTrace.Info("TestRequestManager.ProcessTestRunAttachments: Test run attachments processing completed.");
                     this.testPlatformEventSource.TestRunAttachmentsProcessingRequestStop();
 
-                    // Post the attachments processing complete event
-                    this.metricsPublisher.Result.PublishMetrics(TelemetryDataConstants.TestAttachmentsProcessingCompleteEvent, requestData.MetricsCollection.Metrics);
+                    // Post the attachments processing complete event.
+                    this.metricsPublisher.Result.PublishMetrics(
+                        TelemetryDataConstants.TestAttachmentsProcessingCompleteEvent,
+                        requestData.MetricsCollection.Metrics);
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public void StartTestSession(
+            StartTestSessionPayload payload,
+            ITestHostLauncher testHostLauncher,
+            ITestSessionEventsHandler eventsHandler,
+            ProtocolConfig protocolConfig)
+        {
+            EqtTrace.Info("TestRequestManager.StartTestSession: Starting test session.");
+
+            if (payload.TestPlatformOptions != null)
+            {
+                this.telemetryOptedIn = payload.TestPlatformOptions.CollectMetrics;
+            }
+
+            var requestData = this.GetRequestData(protocolConfig);
+
+            if (this.UpdateRunSettingsIfRequired(
+                payload.RunSettings,
+                payload.Sources,
+                null,
+                out string updatedRunsettings))
+            {
+                payload.RunSettings = updatedRunsettings;
+            }
+
+            if (InferRunSettingsHelper.AreRunSettingsCollectorsInCompatibleWithTestSettings(payload.RunSettings))
```

Comment:
nit: incompatible should be a single word

Reply:
Done.